[BUG] Wrong data type

In the commit 22ef9b2887f13ffab1109573925182aeaf0e4f1, the function `std::string Filesystem::unique_path(string_view model)` under Windows uses `std::wstring` as the data type for the variables `modelStr` and `name`. The library cannot be built because the function `exists` cannot be called in this way!
https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/dd9eb9b6b836d5e2967c222158c865ecaf7b9266/src/libutil/filesystem.cpp#L526

Compiler error MSVC:
```
filesystem.cpp(526,14): error C2664: "bool OpenImageIO_v2_6_1::Filesystem::exists(OpenImageIO_v2_6_1::string_view) noexcept" : 
 Conversion of argument 1 from "std::wstring" to "OpenImageIO_v2_6_1::string_view" not possible
```

--- a/src/libutil/filesystem.cpp
+++ b/src/libutil/filesystem.cpp
@@ -523,7 +523,11 @@ Filesystem::unique_path(string_view model)
         for (size_t i = 0, e = modelStr.size(); i < e; ++i)
             if (name[i] == '%')
                 name[i] = chrs[pick(rg)];
+#    if defined(_WIN32)
+        if (!exists(Strutil::utf16_to_utf8(name)))
+#    else
         if (!exists(name))
+#    endif
             break;
     }
 #    if defined(_WIN32)

